src/elfpy/pricing_models.py

```diff
@@ -732,13 +721,30 @@ def calc_in_given_out(
             The fee the user pays. The units are always in terms of bonds or
             base.
         """
-
+        assert out > 0, f"pricing_models.calc_in_given_out: ERROR: expected out > 0, not {out}!"
+        assert (
+            share_reserves > 0
+        ), f"pricing_models.calc_in_given_out: ERROR: expected share_reserves > 0, not {share_reserves}!"
+        assert (
+            bond_reserves > 0
+        ), f"pricing_models.calc_in_given_out: ERROR: expected bond_reserves > 0, not {bond_reserves}!"
+        assert (
+            1 >= fee_percent >= 0
+        ), f"pricing_models.calc_in_given_out: ERROR: expected 1 >= fee_percent >= 0, not {fee_percent}!"
+        assert (
+            1 > time_remaining >= 0
+        ), f"pricing_models.calc_in_given_out: ERROR: expected 1 > time_remaining >= 0, not {time_remaining}!"
+        assert share_price >= init_share_price >= 1, (
+            f"pricing_models.calc_in_given_out: ERROR:"
+            f" expected share_price >= init_share_price >= 1, not share_price={share_price}"
+            f" and init_share_price={init_share_price}!"
+        )
```

Comment:
👍. Wdyt about using `pricing_models.hyperdrive.calc_in_given_out` so that we can disambiguate between the two pricing models errors? I can change this in my PR if you like this idea.

Reply:
Yup sounds good to me.